qlty-config/src/config/plugin.rs

```diff
@@ -113,6 +113,8 @@ pub struct DriverDef {
 
     #[serde(default)]
     pub missing_output_as_error: bool,
+
+    pub supported_platforms: Option<Vec<Platform>>,
```

Comment:
This use of `Option<Vec<T>>` catches my eye.

By using Option and Vec you can differentiate between null/omitted and an empty array. If that distinction is importanr, then this is the right type.

OTOH, if you want to interpret empty array just like an omitted field (which seems fine to me), then you can just use `Vec<T>` which defaults to `[]` and check it with `is_empty()`